src/main/java/synapseawsconsolelogin/Auth.java

```diff
@@ -316,17 +318,26 @@ public String getAWSSecretKey() {
 		}
 	}
 	
-	private static Properties properties = null;
-
 	public static void initProperties() {
 		if (properties!=null) return;
 		properties = new Properties();
+		
+		String propertyFileName = System.getProperty(PROPERTIES_FILENAME_PARAMETER);
+		if (StringUtils.isEmpty(propertyFileName)) {
+			propertyFileName = System.getenv(PROPERTIES_FILENAME_PARAMETER);
+			
+		}
+		if (StringUtils.isEmpty(propertyFileName)) {
+			propertyFileName = "global.properties";
+			
+		}
+		
```

Comment:
any possibility of a unit test for this?

Reply:
Some testing is possible but you can't, for example, set environment variables in Java.

Reply:
There are now several tests (see below).  This necessitated making the property-related code non-static leading to a bunch of other changes.  (Overall I'd say the code style is improved.)